### PR TITLE
Add px4 ros2 navigation interface

### PR DESCRIPTION
**Pull request by GuillaumeLaine**
Adds docs for PX4 ROS 2 Navigation Interface.

### Restructure
Previously:
```
Summary
├──   ...
└── PX4 ROS 2 Interface Library            # control interface
```
Now:
```
Summary
├──   ...
└── PX4 ROS 2 Interface Library            # parent doc
    ├── PX4 ROS 2 Navigation Interface     # new navigation interface docs
    └── PX4 ROS 2 Control Interface        # control interface moved here
```

Waiting on:
- [x] https://github.com/Auterion/px4-ros2-interface-lib/pull/15

### CODE REVIEW

**Comment by GuillaumeLaine** [comment]
@DanMesh tagging you for early feedback

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:11]: We use first letter capitalization. Minor connecting words can stay lower case.

```suggestion
## Example and First Test
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:104]:
```suggestion
### Local Position Updates
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:181]:
```suggestion
### Global Position Updates
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:1]: I wonder if we might want to use "Navigation Control". Navigation on its own could be telemetry? Or Position Control?

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:7]: It doesn't "matter, but I quite like sentence line breaks. They make reviewing and commenting easier.

```suggestion
The PX4 ROS 2 Interface Library for navigation enables developers to send their position estimates to PX4 directly using ROS 2.
The interface provides a layer of abstraction from PX4 and the uORB messaging framework, and introduces a few sanity checks on the requested state estimation updates sent via the interface.
These estimates are then fused into the EKF as internal PX4 estimates would.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:9]: I'd probably expand on the use cases here that you envisage - presumably providing position via computer vision, or allowing a mission computer that has a GPS to provide that. 

Also do you need to do anything in PX4 - i.e. turn off other sources?

Comment (hamishwillee) [reply en/ros2/px4_ros2_navigation_interface.md:9]: I guess I'm also thinking about the "parent" library, which is mostly about managing offboard control modes such that they have an identity and can safely coexist. The question being, how this related to that or whether it is fully independent.

**Comment by hamishwillee** [comment]
It looks useful as is, but let me know when you're ready for a proper review. 

Minimally you will need some introductory text and a link in the parent doc.

We might also do some more cross linking to reference docs in doxygen once Beat has finished that.

**Comment by GuillaumeLaine** [comment]
@hamishwillee Thanks a lot for the early feedback! I'll let you know when it's ready for a prroper review :+1:

File: en/ros2/px4_ros2_navigation_interface.md
Comment (DanMesh) [inline en/ros2/px4_ros2_navigation_interface.md:68]:
```suggestion
7. In the PX4 shell, you can check that PX4 receives global position updates:
```

**Review by DanMesh — DISMISSED** [review]
Looking good!

I had one or two questions to think about, and a few minor language suggestions.

File: en/ros2/px4_ros2_navigation_interface.md
Comment (GuillaumeLaine) [reply en/ros2/px4_ros2_navigation_interface.md:58]: Good idea, I'll rephrase a bit so it makes sense

File: en/ros2/px4_ros2_control_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_control_interface.md:14]: From here, best to link to the parent page:

```suggestion
The [PX4 ROS 2 Interface Library](ros2/px4_ros2_interface_lib.md) is a C++ library that simplifies controlling PX4 from ROS 2.
```

File: en/ros2/px4_ros2_control_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_control_interface.md:25]: Maybe?

```suggestion
This diagram provides a conceptual overview of how the control interface modes and mode executors interact with PX4.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:7]:
```suggestion
The [PX4 ROS 2 Interface Library](../ros2/px4_ros2_interface_lib.md) navigation interface enables developers to send their position measurements to PX4 directly from ROS 2 applications, such as a VIO system or a map matching system.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:9]:
```suggestion
These measurements are then fused into the EKF just as though they were internal PX4 measurements.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:11]:
```suggestion
The library provides two classes, [`LocalPositionMeasurementInterface`](https://auterion.github.io/px4-ros2-interface-lib/classpx4__ros2_1_1LocalPositionMeasurementInterface.html) and [`GlobalPositionMeasurementInterface`](https://auterion.github.io/px4-ros2-interface-lib/classpx4__ros2_1_1GlobalPositionMeasurementInterface.html), which both expose a similar `update` method to provide either a local position or global position update to PX4, respectively.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:12]:
```suggestion
The `update` method expects a position measurement `struct` ([LocalPositionMeasurement](https://auterion.github.io/px4-ros2-interface-lib/structpx4__ros2_1_1LocalPositionMeasurement.html) or [`GlobalPositionMeasurement`](https://auterion.github.io/px4-ros2-interface-lib/structpx4__ros2_1_1GlobalPositionMeasurement.html)) which developers can populate with their own generated position measurements.
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:14]: Just to be consistent in naming

```suggestion
## Installation and First Test
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:57]: Whats going on here? i.e. looks like you ar running it two times example_global_navigation_cpp ? Is this first the package to build and then the executable to run or something?

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:53]:
```suggestion
6. Back in the ROS 2 terminal, source the workspace you just built (in step 3) and run the [global_navigation](https://github.com/Auterion/px4-ros2-interface-lib/tree/main/examples/cpp/navigation/global_navigation) example which periodically sends dummy global position updates:
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:97]: The downside of using first letter of each word capitalization, is that it isn't always clear which letters you can lower case.

```suggestion
## How to use the Library
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:99]:
```suggestion
To send a position measurement, you populate the position struct with the values you have measured.
```

File: en/ros2/px4_ros2_control_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_control_interface.md:191]:
```suggestion
## How to use the Library
```

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:107]:
```suggestion
For a basic example of how to use this interface, check out the [examples](https://github.com/Auterion/px4-ros2-interface-lib/tree/main/examples/cpp/navigation) in the `Auterion/px4-ros2-interface-lib` repository, such as [examples/cpp/navigation/local_navigation](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/examples/cpp/navigation/local_navigation/include/local_navigation.hpp) or [examples/cpp/navigation/global_navigation](https://github.com/Auterion/px4-ros2-interface-lib/blob/main/examples/cpp/navigation/local_navigation/include/global_navigation.hpp).
```

File: en/ros2/px4_ros2_control_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_control_interface.md:120]: FWIW all this is now duplicated in three places. I think that's OK, but if we end up with too many of these we might instead link up for this.

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:11]: FWIW I always link to first instance (at least) of the name of the interface.

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:117]: I've inconsistently updated `update()`. When talking about methods I prefer having `()` after the name because then it is obvious you're talking about a method. Some people don't like this because they think the empty bracket indicates "no arguments".

File: en/ros2/px4_ros2_navigation_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_navigation_interface.md:300]: FYI I moved this note from the "How to use" intro to its own section. It didn't feel right as a note.

**Comment by hamishwillee** [comment]
@GuillaumeLaine Yes, the new structure is much better. I'm happy to merge this if the corresponding development has all been done?

**Review by hamishwillee — APPROVED** [review]

File: en/ros2/px4_ros2_navigation_interface.md
Comment (GuillaumeLaine) [reply en/ros2/px4_ros2_navigation_interface.md:57]: Good catch! Exactly, the ros2 run syntax is `ros2 run <package_name> <executable_name>`

This command was currently functional, but I think it's bad practice to have the package and executable names the exact same.

I'm going to rename the executable `example_global_navigation_cpp` -> `example_global_navigation` in the implementation and the docs. This will then reflect the command used to run the mode examples.

**Comment by GuillaumeLaine** [comment]
@hamishwillee Thanks a lot for the feedback and reviews! I'm just waiting for the [executable rename PR](https://github.com/Auterion/px4-ros2-interface-lib/pull/15) to get in, otherwise corresponding development is all done. I'll ping you here for merging once that other PR gets in :+1:

**Comment by hamishwillee** [comment]
Excellent thanks. This is approved now, so anyone with merge rights can push it in (though I will do this as soon as I see a message from you).

File: en/ros2/px4_ros2_control_interface.md
Comment (hamishwillee) [inline en/ros2/px4_ros2_control_interface.md:1]: FYI @GuillaumeLaine I just added this `Badge` to the headings in the three docs. Part of a plan to make it clear the versions in which stuff is known to work.

Note I'd rather do this using page front metadata, but that's going to require some more thinking.

**Comment by GuillaumeLaine** [comment]
@hamishwillee The associated PR has been merged, so feel free to merge this PR whenever you're ready! Thanks again!

**Comment by hamishwillee** [comment]
Excellent! Thanks for merging @DanMesh

**Pull request merged**
